packages/edit-site/src/components/layout/index.js

```diff
@@ -116,7 +116,7 @@ export default function Layout() {
 	} );
 	const disableMotion = useReducedMotion();
 	const showSidebar =
-		( isMobileViewport && ! isListPage ) ||
+		( isMobileViewport && canvasMode === 'view' && ! isListPage ) ||
 		( ! isMobileViewport && ( canvasMode === 'view' || ! isEditorPage ) );
```

Comment:
Could this logic be simplified? The checks in both instances are pretty similar now, and `isEditorPage` is defined as: `const isEditorPage = ! isListPage;`. On larger screens, the check `canvasMode === 'view'` is enough to get it to be `true`. Maybe all we need is `canvasMode === 'view'?

Reply:
Thanks for the review!

> Maybe all we need is `canvasMode === 'view'`?

I tried to find out why each viewport has a condition, and why `isEditorPage` and `isListPage` are included in the conditions.

One reason seems to be to properly control the sidebar on the Pattern page.

On the Pattern page, `canvasMode` is always `view` regardless of the viewport. However, in the mobile viewport, when a pattern category is selected, the sidebar must be hidden even if the `canvasMode` is `view`. Otherwise, the two regions, sidebar and pattern content, will be rendered overlapping, as shown in the screenshot below.

I believe that avoiding this is the reason for such complicated conditions.

![navigatable-region](https://github.com/WordPress/gutenberg/assets/54422211/c1ab8451-6780-4256-9c2f-7e6a97989cb6)

Reply:
Ah, thanks for looking into this! I played around with it as well, and _I think_ we could remove the `! isEditorPage` from the not mobile conditional, but since that isn't related to this PR it could be a follow-up if you also think it should be removed.